rfcs/0000-api-cr-crowding.md

```diff
@@ -0,0 +1,337 @@
+  `:not_accepting_passengers`, but for the Keolis data will only use 3 of them, with the mapping
+  based on `MedianDensityFlag` of `0` to `:many_seats_available`, `1` to `:few_seats_available`,
+  and `2` to `:full` (unless dotcom already has a different mapping of GTFS terms to bowling
+  pins?).
+- `:occupancy_percentage` is just the `MedianDensity` value verbatim.
+- `:valid_days` is the list `[1, 2, 3, 4, 5, 6, 7]` representing this data applying to all days of
+  the week (akin to how `Service` works in the API).
+- `:start_date` is the current date
+- `:end_date` is the date two weeks from today
+- `:start_time` is `nil`.
+
+The index (/indices) will depend on what sort of filtering (if any) we provide, based on above.
+
+## Fetching the data
+
+The Keolis endpoint should be fetched every ten minutes. We could use with modifications the
+existing `StateMediator` infrastructure to do this. State mediators can be configured to fetch
+from a URL on a periodic basis, but they take `url` as a string configuration, while the URL for
+the firebase feed changes periodically because it has a `?token=....` parameter, and that token is
+only valid for a short while. So `StateMediator.Mediator` should be updated to optionally take a
+MFA tuple for its URL configuration as well.
+
+From futzing around in a Livebook, some code to download the data from Keolis is:
+
+```ex
+Mix.install([:jason, :hackney, goth: "1.3.0-rc.3"])
+firebase_url = "..."
+credentials_json = "..."
+credentials = Jason.decode!(credentials_json)
+
+scopes = [
+  "https://www.googleapis.com/auth/firebase.database",
+  "https://www.googleapis.com/auth/userinfo.email"
+]
+
+source = {:service_account, creds, scopes: scopes}
+{:ok, pid} = Goth.start_link(name: MyGoth, source: source)
+{:ok, token} = Goth.fetch(MyGoth)
+resp = :hackney.request(firebase_url <> ".json?access_token=" <> token.token)
+{:ok, 200, _, ref} = resp
+{:ok, data} = :hackney.body(ref)
+Jason.decode!(data)
+```
+
+The `Mediator` can invoke `new_state` on its configured module (`State.Occupancy`), which will
+then do the mapping and merging of the data as specified above.
+
+## API controller
+
+Fill out the swagger docs, but make sure it says **EXPERIMENTAL** and "can change without prior
+warning".
+
+The `includes` and `filters` sections depend on what is decided above, about what is necessary for
+dotcom.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+A drawback of building on GTFS-Occupancies at this point, is that it's still in flux and can
+change in the future. But we think that it's better to build on what's there, which will hopefully
+be close to the final product (it's gone through lots of revisions at this point and seems to be
+settling down) rather than do our own thing and then end up with a totally different way of
+handling this data in the future.
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Actual GTFS
```

Comment:
What about an approach where an app generates an `occupancies.txt` separately from our GTFS, and the API consumes that file directly?